### PR TITLE
support debugging execution plan generation

### LINKED ISSUE #940: Feature request: Support debugging execution
### Similar requests

- [X] I have searched and found no existing similar requests

### How are you using Studio?

_No response_

### What problems are you trying to solve?

It would be extremely helpful to be able to debug execution, _especially when we're still in the process of making execution error message more descriptive_. So next to every place that we allow execute, we should also allow debug, basically, it would print out the steps in the execution so when it stops somewhere, we could trace the step at which the problem occurred.

### Describe the solution you would like to see implemented

- [x] Integrate `engine` API for debug execution - @pierredebelen - https://github.com/finos/legend-engine/pull/596
- [x] Create an UI component for execute button, this button basically will have the word execute on it with a dropdown next to it to allow users to either create a plan or to debug execution

### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [ ] I would like to work on this feature

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] Closes #940 
- [x] Allow turning off network request payload compression for better debugging in `Legend Query`

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/13574879/160334494-0592c972-cb5b-4d17-ac35-6300d3fb2d65.mov

### CODE REVIEW

File: packages/legend-query/src/components/QueryBuilder.tsx [hunk 1]
```diff
@@ -208,107 +218,149 @@ export const QueryBuilder = observer(
...
-                            <QueryBuilderFunctionsExplorerPanel
-                              queryBuilderState={queryBuilderState}
-                            />
-                          </ResizablePanel>
-                        )}
-                        {queryBuilderState.showFunctionPanel &&
-                        queryBuilderState.showParameterPanel ? (
-                          <ResizablePanelSplitter />
+            <div className="query-builder__activity-bar">
+              <div className="query-builder__activity-bar__items"></div>
```
Comment (MauricioUyaguari) [inline packages/legend-query/src/components/QueryBuilder.tsx:222]: Can we add a changelog message for this. Important feature for debugging :)

Comment (akphi) [reply packages/legend-query/src/components/QueryBuilder.tsx:222]: of course!

File: packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingExecutionState.ts [hunk 2]
```diff
@@ -679,13 +685,40 @@ export class MappingExecutionState {
...
+              this.mappingEditorState.mapping,
+              query,
+              runtime,
+              PureClientVersion.VX_X_X,
+            )) as { plan: RawExecutionPlan; debug: string };
+          rawPlan = debugResult.plan;
+          this.executionPlanState.setDebugText(debugResult.debug);
+          this.executionPlanState.setShowDebugPanel(true);
+        } else {
+          rawPlan =
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingExecutionState.ts:702]: same comment as for service

File: packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts [hunk 3]
```diff
@@ -700,16 +701,43 @@ export class MappingTestState {
...
+            this.mappingEditorState.mapping,
+            this.queryState.query,
+            this.inputDataState.runtime,
+            PureClientVersion.VX_X_X,
+          )) as { plan: RawExecutionPlan; debug: string };
+        rawPlan = debugResult.plan;
+        this.executionPlanState.setDebugText(debugResult.debug);
+        this.executionPlanState.setShowDebugPanel(true);
+      } else {
+        rawPlan =
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts:721]: same comment as for service

Comment (akphi) [reply packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts:721]: that's a fair point, I actually did this for `Legend Query`, I think I can skim down that.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
reviewed

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**